src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -4,24 +4,47 @@
  */
 package com.vmware.vipclient.i18n.messages.api.opt.local;
 
+import java.util.Map;
+
 import org.json.simple.JSONObject;
 
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.base.cache.MessageCacheItem;
+import com.vmware.vipclient.i18n.messages.api.opt.MessageOpt;
 import com.vmware.vipclient.i18n.messages.api.opt.Opt;
 import com.vmware.vipclient.i18n.messages.dto.MessagesDTO;
+import com.vmware.vipclient.i18n.util.FormatUtils;
 import com.vmware.vipclient.i18n.util.JSONBundleUtil;
 
-public class LocalMessagesOpt implements Opt {
+public class LocalMessagesOpt implements Opt, MessageOpt {
+	
+	private static final String OFFLINE_RESOURCE_PATH = "{0}/messages_{1}.json";
     private MessagesDTO dto;
 
     public LocalMessagesOpt(MessagesDTO dto) {
         this.dto = dto;
     }
 
+    @Deprecated
     public JSONObject getComponentMessages() {
         return JSONBundleUtil.getMessages(dto.getLocale(), dto.getProductID(),
                 dto.getVersion(), dto.getComponent());
     }
+    
+    @Override
+    public void getComponentMessages(MessageCacheItem cacheItem) {
```

Comment:
how about using 'refreshComponentMessages' as the function name?

Reply:
but it returns 'void', meaning outside caller get nothing. If there's case to detect the change on service, how to do it?

Reply:
Caller of the method gets the change inside the object that is passed as argument.

"All object references in Java are passed by value. This means that a copy of the value will be passed to a method. Passing a copy of the value also changes the real value of the object. "

What do you mean by "case to detect the change on service, how to do it"?